### DISCUSSION TITLE
Electrolama ZZH (CC2652R stick) - doesn't join zigbee network as router.

### DISCUSSION THREAD

**Discussion by Pansovi4** [Koenkk/zigbee2mqtt#11566] [Devices]
Hello all,

I recently got Sonoff Zigbee 3.0 Dongle plus (firmware: 20210708).
I am using zigbee2mqtt with Electrolama ZZH (CC2652R Stick).

My plan was to replace coordinator to Sonoff stick and make ZZH stick as router.
I changed sticks - Sonoff dongle is up and running, all devices stayed connected and working ok.

Next plan to make ZZH as router. I flashed ZZH stick with router software with FlashProgrammer.
Now when I plug this stick in - i can’t see it in zigbee network - it doesn’t try to join network.

My question - what am i doing wrong? Is there any additional actions needed? I have 23 devices in zigbee network, couple relays working as router, i’ve never added stick as router before.
Will it autojoin or do i need to make it somehow in pairing mode?

**Comment by Pansovi4** [comment]
That's what I got in LOG.
Looks like Zigbee2mqtt remember ZZH stick as old coordinator and *reserved* name - coordinator for it???

Error Error: Cannot read property 'endpoints' of undefined
Error Error: Cannot read property 'endpoints' of undefined
Error Error: Cannot read property 'endpoints' of undefined
Error Error: Cannot read property 'endpoints' of undefined
Error Error: Cannot read property 'endpoints' of undefined
Error Error: Cannot read property 'endpoints' of undefined
Info Device 'Coordinator' joined
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"Coordinator","ieee_address":"0x00124b00216000ce"},"type":"device_joined"}'
Info Configuring 'Coordinator'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"Coordinator"},"type":"device_connected"}'
Info MQTT publish: topic 'homeassistant/sensor/0x00124b00216000ce/linkquality/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_undefined"],"manufacturer":"Custom devices (DiY)","model":"Texas Instruments router (ti.router)","name":"Coordinator","sw_version":"20210128"},"enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:signal","json_attributes_topic":"zigbee2mqtt/Coordinator","name":"Coordinator linkquality","state_class":"measurement","state_topic":"zigbee2mqtt/Coordinator","unique_id":"undefined_linkquality_zigbee2mqtt","unit_of_measurement":"lqi","value_template":"{{ value_json.linkquality }}"}'
Info MQTT publish: topic 'homeassistant/sensor/0x00124b00216000ce/last_seen/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"}],"device":{"identifiers":["zigbee2mqtt_undefined"],"manufacturer":"Custom devices (DiY)","model":"Texas Instruments router (ti.router)","name":"Coordinator","sw_version":"20210128"},"device_class":"timestamp","enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:clock","json_attributes_topic":"zigbee2mqtt/Coordinator","name":"Coordinator last seen","state_topic":"zigbee2mqtt/Coordinator","unique_id":"undefined_last_seen_zigbee2mqtt","value_template":"{{ value_json.last_seen }}"}'
Info MQTT publish: topic 'zigbee2mqtt/Coordinator', payload '{"last_seen":"2022-02-22T16:07:47.851Z","linkquality":null}'
Info Starting interview of 'Coordinator'
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"Coordinator","ieee_address":"0x00124b00216000ce","status":"started"},"type":"device_interview"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"Coordinator"},"type":"pairing"}'
Info MQTT publish: topic 'zigbee2mqtt/Coordinator', payload '{"last_seen":"2022-02-22T16:07:47.973Z","linkquality":null}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"Coordinator","ieee_address":"0x00124b00216000ce"},"type":"device_announce"}'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"Coordinator"},"type":"device_announced"}'
Info MQTT publish: topic 'homeassistant/sensor/0x00124b00216000ce/linkquality/config', payload 'null'
Info MQTT publish: topic 'homeassistant/sensor/0x00124b00216000ce/last_seen/config', payload 'null'
Info MQTT publish: topic 'zigbee2mqtt/Вода туалет ', payload '{"battery":100,"battery_low":null,"last_seen":"2022-02-22T16:07:49.142Z","linkquality":134,"temperature":35,"voltage":3015,"water_leak":false}'
Error Failed to configure 'Coordinator', attempt 1 (Error: Bind 0x00124b00216000ce/8 genBasic from '0x00124b00216000ce/1' failed (AREQ - ZDO - bindRsp after 10000ms) at Timeout._onTimeout (/app/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35) at listOnTimeout (internal/timers.js:557:17) at processTimers (internal/timers.js:500:7))
Info Successfully interviewed 'Coordinator', device has successfully been paired
Info Device 'Coordinator' is supported, identified as: Custom devices (DiY) Texas Instruments router (ti.router)
Info MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Texas Instruments router","exposes":[{"access":1,"description":"Link quality (signal strength)","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"ti.router","options":[],"supports_ota":false,"vendor":"Custom devices (DiY)"},"friendly_name":"Coordinator","ieee_address":"0x00124b00216000ce","status":"successful","supported":true},"type":"device_interview"}'
Info Configuring 'Coordinator'
Info MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"description":"Texas Instruments router","friendly_name":"Coordinator","model":"ti.router","supported":true,"vendor":"Custom devices (DiY)"},"type":"pairing"}'
Info MQTT publish: topic 'zigbee2mqtt/Coordinator', payload '{"last_seen":"2022-02-22T16:08:11.830Z","linkquality":138}'
Info Successfully configured 'Coordinator'

**Comment by Hedda** [comment]
Try clear/reset NVRAM with zigpy-znp, change extended_pan_id to dddddddddddddddd, and edit database, as described here -> https://github.com/Koenkk/zigbee2mqtt/issues/9117

> Just my two cents if somebody finds this.., clearing the nvram only did not work for me but I did the below steps, but I am not sure what step really did the fix:
> 
> * stop zigbee2mqtt
> * Do the nvram clear
> * unplug the stick, and re-add (did you turn it off and on??)
> * Change extended_pan_id to dddddddddddddddd
> * in database.db, I removed all coordinators, for some reason I had multiple
> * start zigbee2mqtt
> 
> After this all pairing did work directly, not even I had to keep items close to the coordinator.
> 
> This was after I upgraded from the good old CC2531 chip to the sonoff zigbee3 CC2652P stick.

**Comment by RenanAz** [comment]
I was able to finally add my ZZH as a router by writing a new secondary Mac to the board (using Flash Programmer 2). Changed the last digit of the actual IEEE and added as secondary. 
Previously was unable to join due to obvious address conflict (since i recently added this device's IEEE as secondary to my Sonoff).
Anyway, thanks for this thread, it did help.
